diesel_cli/src/infer_schema_internals/inference.rs

```diff
@@ -109,6 +109,12 @@ pub fn load_foreign_key_constraints(
 
     constraints.map(|mut ct| {
         ct.sort();
+        ct.iter_mut().for_each(|foreign_key_constraint| {
+            if RESERVED_NAMES.contains(&foreign_key_constraint.foreign_key.as_str()) {
```

Comment:
This check could be done in `impl<'a> Display for Joinable<'a> ` alternatively, but this way the field is quite explicit.